src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -745,51 +774,60 @@ public List<Consumer> getGuests(Consumer consumer) {
      */
     @Transactional
     public Consumer getHypervisor(String hypervisorId, Owner owner) {
-        return (Consumer) currentSession().createCriteria(Consumer.class)
-            .add(Restrictions.eq("ownerId", owner.getId()))
-            .createAlias("hypervisorId", "hvsr")
-            .add(Restrictions.eq("hvsr.hypervisorId", hypervisorId.toLowerCase()))
-            .setMaxResults(1)
-            .uniqueResult();
+        String jpql = """
+            SELECT c FROM Consumer c
+            WHERE c.ownerId = :ownerId
+                AND LOWER(c.hypervisorId.hypervisorId) = :hypervisorId
+            """;
+
+        try {
+            return getEntityManager().createQuery(jpql, Consumer.class)
+                .setParameter("ownerId", owner.getId())
+                .setParameter("hypervisorId", hypervisorId.toLowerCase())
+                .setMaxResults(1)
+                .getSingleResult();
+        }
+        catch (NoResultException e) {
+            return null;
+        }
     }
 
     @Transactional
     public Consumer getExistingConsumerByHypervisorIdOrUuid(String ownerId, String hypervisorId,
         String systemUuid) {
-        VirtConsumerMap hypervisorMap = new VirtConsumerMap();
-        Consumer found = null;
-        String sql = "select consumer_id from cp_consumer_hypervisor " +
-            "where hypervisor_id = :hypervisorId " +
-            "and owner_id = :ownerId";
+        String jpql = """
+            SELECT h.consumer.id FROM HypervisorId h
+            WHERE h.hypervisorId = :hypervisorId
+                AND h.owner.id = :ownerId
+            """;
 
-        Query query = this.currentSession()
-            .createSQLQuery(sql)
+        List<String> consumerIds = getEntityManager().createQuery(jpql, String.class)
             .setParameter("ownerId", ownerId)
-            .setParameter("hypervisorId", hypervisorId.toLowerCase());
-        List<String> consumerIds = query.list();
+            .setParameter("hypervisorId", hypervisorId.toLowerCase())
+            .getResultList();
 
-        if (consumerIds != null && consumerIds.size() > 0) {
-            List<String> one = Collections.singletonList(consumerIds.get(0));
-            found = (Consumer) ((List) this.getConsumers(one)).get(0);
+        Consumer found = null;
+        if (!consumerIds.isEmpty()) {
+            found = getEntityManager().find(Consumer.class, consumerIds.get(0));
```

Comment:
Same with this bit of logic here: we're blindly hoping that the query returned only a single result and that it's our desired result and then going through a bit of clunk to actually get to the result. We could just as easily return the consumer straight away.

This particular method is doubly offensive in that rather than stacking the two operations logically at the business or logic layer, we're combining them into one here in a way that's not clear how it'll do the lookup due to the lack of documentation or any conveyance in how anything is defined or named.

Really this whole thing could be two methods: `getConsumerByHypervisorId` (which looks to be what `getHypervisor` is), and `getConsumerBySystemUuid`, and then the logic in `HypervisorUpdateAction` cleaned up to only do the second check if the consumer from the first check is null and the config (which should be pulled when the object is built! Ahhh!) is set accordingly.

I know this isn't the fault of your changes here, but techdebt stuff like this should never just be about blindly porting A to B without taking a moment to fix some of the more wild bits like this which need some attention.

Reply:
Yes , absolutely true. Moved.